lib/src/diff.rs

```diff
@@ -71,6 +75,73 @@ pub fn find_nonword_ranges(text: &[u8]) -> Vec<Range<usize>> {
         .collect()
 }
 
+/// Compares byte sequences based on a certain equivalence property.
+pub trait CompareBytes {
```

Comment:
Nit: I would add a comment here, _why_ we are generic instead of a wrapper type or a taking a callback. So noone sweeps in trying to "improve" it.

Reply:
Good point. Expanded doc comment.